framework/src/base/MooseApp.C

```diff
@@ -1709,6 +1709,20 @@ MooseApp::hasRestartableMetaData(const std::string & name,
   }
 }
 
+RestartableDataValue &
+MooseApp::getRestartableMetaData(const std::string & name,
+                                 const RestartableDataMapName & metaname) const
+{
```

Comment:
so if this is queried in a threaded region, it will fail because the map is not well protected. Let's add a thread id argument and add an error just like in registerRestartableData

Reply:
Added a thread ID argument and error condition